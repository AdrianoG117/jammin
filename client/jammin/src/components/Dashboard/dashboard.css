.dashboard-container {
  /* background-color: green; */
  width: 100%;
  height: 90%;
  display: flex;
  flex-direction: column;
  align-items: center;
}

.myevents-list {
  /* background-color: yellow; */
  width: 70%;
  height: 70%;
  margin-top: 3rem;
  overflow-x: scroll;
  display: flex;
  flex-direction: column;
  align-items: center;
}
