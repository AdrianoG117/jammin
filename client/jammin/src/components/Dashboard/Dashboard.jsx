import React, { useEffect } from 'react';
import MyJamsItem from '../MyJamsItem/MyJamsItem';
import './dashboard.css';

function Dashboard({ userData, setUserData }) {
  useEffect(() => {
    //sort events
    userData.comingEvents.sort(function (a, b) {
      return a.date < b.date ? -1 : a.date > b.date ? 1 : 0;
    });
  }, []);

  return (
    <div className="dashboard">
      {userData.comingEvents.length ? (
        <div className="dashboard-container">
          <h1>Hi there {userData.firstname} !</h1>
          <h2>Check out your upcoming events below: </h2>
          <div className="myevents-list">
            {userData.comingEvents.length
              ? userData.comingEvents.map((event) => (
                  <MyJamsItem
                    key={event._id}
                    eventData={event}
                    userData={userData}
                    setUserData={setUserData}
                  />
                ))
              : null}
          </div>
        </div>
      ) : (
        <div className="dashboard-container">
          <h1>Hi {userData.firstname}</h1>
          <h2>You have no jams added yet, go to the Find Jam section to find events around you</h2>
        </div>
      )}
    </div>
  );
}

export default Dashboard;
