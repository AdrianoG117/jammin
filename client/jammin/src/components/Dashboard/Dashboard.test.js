import {render, screen} from '@testing-library/react';
import Dashboard from './Dashboard';


describe('Topbar component', ()=> {
  
  test('should match the snapshot', () => {
    const {container} = render(<Dashboard
                  setUserData={()=>{}}
                  setIsSignedUp={()=>{}} />);
    expect(container.firstChild).toMatchSnapshot();
  })

  test('Should render two titles when user has coming events', () => {
    render (<Dashboard userData={true}
        setUserData={()=>{}}
        setIsSignedUp={()=>{}} />);
    //title
    screen.getByText(/Hi there {}/)
    screen.getByText(/Check out your coming events below: /)
    })
    // also need to add testing to see MyJamsItem is rendered

  test('Should render two titles when user has no coming events', () => {
    render (<Dashboard userData={false}
        setUserData={()=>{}}
        setIsSignedUp={()=>{}} />);
    //title
    screen.getByText(/Hi {}/)
    screen.getByText(/You have no jams added yet, go to the Find Jam section to find events around you/)
    })
})

