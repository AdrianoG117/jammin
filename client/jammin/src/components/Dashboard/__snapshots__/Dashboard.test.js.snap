// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component should match the snapshot with array 1`] = `
<div
  class="dashboard"
>
  <div
    class="dashboard-container"
  >
    <h1>
      Hi there 
      ad
       !
    </h1>
    <h2>
      Check out your upcoming events below: 
    </h2>
    <div
      class="myevents-list"
    >
      <div
        class="jamsitem-container"
      >
        <div
          class="date"
        >
          <h2>
            Oct 22nd
          </h2>
        </div>
        <div
          class="item-main"
        >
          <h1>
            Test1
          </h1>
          <h2>
            Miami, FL, USA
          </h2>
        </div>
        <div
          class="button-container"
        >
          <button
            id="see-btn"
          >
            See Event
          </button>
        </div>
        <div
          class="trash-container"
          role="button"
          tabindex="0"
        >
          <img
            alt=""
            src="trash.png"
          />
        </div>
      </div>
      <div
        class="jamsitem-container"
      >
        <div
          class="date"
        >
          <h2>
            Oct 22nd
          </h2>
        </div>
        <div
          class="item-main"
        >
          <h1>
            Test1
          </h1>
          <h2>
            Miami, FL, USA
          </h2>
        </div>
        <div
          class="button-container"
        >
          <button
            id="see-btn"
          >
            See Event
          </button>
        </div>
        <div
          class="trash-container"
          role="button"
          tabindex="0"
        >
          <img
            alt=""
            src="trash.png"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
